datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SecurityAccessManifester.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.datalake.service.sdx;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.springframework.stereotype.Service;
+
+import com.google.common.base.Strings;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.instancegroup.InstanceGroupV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.instancegroup.securitygroup.SecurityGroupV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.util.requests.SecurityRuleV4Request;
+import com.sequenceiq.cloudbreak.util.CidrUtil;
+import com.sequenceiq.common.api.type.InstanceGroupType;
+
+@Service
+public class SecurityAccessManifester {
+
+    @Inject
+    private InternalApiCallCalculator internalApiCallCalculator;
+
+    public void overrideSecurityAccess(InstanceGroupType instanceGroupType, List<InstanceGroupV4Request> instanceGroups, String securityGroupId, String cidrs) {
+        instanceGroups.stream()
+                .filter(ig -> ig.getType() == instanceGroupType)
+                .findFirst()
+                .ifPresent(ig -> {
+                    SecurityGroupV4Request securityGroup = ig.getSecurityGroup();
+                    if (securityGroup == null) {
+                        securityGroup = new SecurityGroupV4Request();
+                    }
+                    if (!internalApiCallCalculator.isInternalApiCall(securityGroup)) {
+                        if (!Strings.isNullOrEmpty(securityGroupId)) {
+                            securityGroup.setSecurityGroupIds(Set.of(securityGroupId));
+                            securityGroup.setSecurityRules(new ArrayList<>());
+                        } else if (!Strings.isNullOrEmpty(cidrs)) {
+                            List<SecurityRuleV4Request> generatedSecurityRules = new ArrayList<>();
+                            List<SecurityRuleV4Request> originalSecurityRules = securityGroup.getSecurityRules();
+                            for (String cidr : CidrUtil.cidrs(cidrs)) {
```

Comment:
This code looks lexactly like the part in `InstanceGroupV1ToInstanceGroupV4Converter.java`.
If so it should be extracted.